esm_parser/yaml_to_dict.py

```diff
@@ -120,6 +122,15 @@ def yaml_file_to_dict(filepath):
                 check_changes_duplicates(yaml_load, filepath + extension)
                 # Add the file name you loaded from to track it back:
                 yaml_load["debug_info"] = {"loaded_from_file": yaml_file.name}
+                if loader.env_variables:
+                    runtime_env_changes = yaml_load.get("general", {}).get("runtime_environment_changes", {})
+                    add_export_vars = runtime_env_changes.get("add_export_vars", [])
+                    for env_var_name, env_var_value in loader.env_variables:
+                        add_export_vars.append(f"{env_var_name}={env_var_value}")
+                    # TODO(PG): There is probably a more elegant way of doing this:
+                    yaml_load['general'] = yaml_load.get("general") or {}
+                    yaml_load['general']['runtime_environment_changes'] = yaml_load['general'].get('runtime_environment_changes') or {}
```

Comment:
I don't understand the syntax here, is the `or` doing the same as `yaml_load.get("general", {})`?

Reply:
Yeah, I had initially thought the same, but apparently `dict.get(a, b)` gives you back a copy instead of a reference, and then modifying the list never shows up in the finalized config. I don't like it either, as it can probably be written down more concisely, but I didn't figure out how....

Reply:
Thanks for the explanation, it's very valuable info!

I don't have a better solution.